generate_translations.mjs

```diff
@@ -1,21 +1,38 @@
 import fs from 'fs';
 import fetch from 'node-fetch';
 import path from 'path';
+import { exit } from 'process';
 
 const translationDir = "joinlemmy-translations/translations/";
 const outDir = "src/shared/translations/";
 const translatorsJsonFile = "lemmy-translations/translators.json";
 const statsFile = "lemmy-instance-stats/stats.json";
+const recommendationsFile = "lemmy-instance-stats/recommended-instances.csv";
```

Comment:
This seems pointlessly overcomplicated, especially bundling recommended into lemmy instance stats. (and mixing csvs and json?)

Why not just hard code a domain list in `instances.tsx` (or utils.tsx), move the filter to that file to build the recommended list. Its NBD to just change that list of recommended domains in the joinlemmy code as things change and we want to recommend different instances. Seems like it should belong here rather than in an instance stats scraper.

IE 

```ts
export const recommendedDomains = ["lemmy.ml", "lemmygrad.ml"....];
...
// filter functions...
```

Reply:
Mainly because i want to start crawling from these recommended instances, and not only from lemmy.ml. This removes an aspect of centralization, but more importantly it prevents problems if a recommended instance is not included in the crawl for some reason.

csv is just simpler in this case, as we only need a simple list of domains, and no complex data.

Reply:
Actually i'm starting to think that the lemmy-instance-stats repo doesnt make much sense. If the goal is to collect statistics, then that should be done at regular intervals, not randomly every 1-4 weeks whenever we happen to update joinlemmy. So i would simply run the crawl as part of the build on ci. That will also make it much easier to make automatic deploys of the site (eg every day).

Keeping statistics should be completely separate from joinlemmy. And the stats  repo isnt used anywhere else afaik, so it seems pretty useless.

Reply:
I agree that the statistics should be separate, and a git submodule for stats seems like the best way to do that. 

But I don't know if I agree with moving this to the CI... I like to be able to test this locally, and run the stats update on my local machine, and be able to predict exactly what the site will look like.

I agree tho as far as keeping historical statistics, its not doing that job unless we throw it in a crontab.